aws-sdk-core/lib/aws-sdk-core/partitions.rb

```diff
@@ -0,0 +1,153 @@
+module Aws
+
+  # A {Partition} is a group of AWS {Region} and {Service} objects. You
+  # can use a partition to determine what services are available in a region,
+  # or what regions a service is available in.
+  #
+  #
+  # ## Partitions
+  #
+  # **AWS accounts are scoped to a single partition**. You can get a partition
+  # by name. Valid partition names include:
+  #
+  # * `"aws"` - Public AWS partition
+  # * `"aws-cn"` - AWS China
+  # * `"aws-us-gov"` - AWS GovCloud
+  #
+  # To get a partition by name:
+  #
+  #     aws = Aws.partition('aws')
+  #
+  # You can also enumerate all partitions:
+  #
+  #     Aws.partitions.each do |partition|
+  #       puts partition.name
+  #     end
+  #
```

Comment:
I'd love the "all everything in everywhere" example to be present, since that's a use case some have. For example:

```ruby
Aws.partitions.each do |partition|
  partition.regions.each do |region|
    region.services.each do |service|
      # Do anything you'd want to do for all services in all regions.
      puts "#{service} is in #{region.name} in #{partition.name}"
    end
  end
end
```